lib/DatabaseCommand.php

```diff
@@ -0,0 +1,127 @@
+<?php
+include("DatabaseConnector.php");
+/**
+ * Class for working with Database
+ *
+ * @author Pavel Vyskočil <vyskocilpavel@muni.cz>
+ */
+class DatabaseCommand
+{
+
+	const WHITELIST = "whiteList";
+	const GREYLIST = "greyList";
+
+	/**
+	 * Function returns array of all IdPs in whitelist/greylist
+	 * @param string $tableName 'whitelist' or 'greylist'
+	 * @return array of all IdPs, every IdP is represents as array
+	 */
+	public static function getAllIdps($tableName) {
+		$databaseConnector = new DatabaseConnector();
+		$conn = $databaseConnector->getConnection();
+		$whiteListTableName = $databaseConnector->getWhiteListTableName();
+		$greyListTableName = $databaseConnector->getGreyListTableName();
+		$table = null;
+		$listOfIdPs = array();
+		assert($conn != NULL);
+
+		if ($tableName == self::WHITELIST) {
+			$table = $whiteListTableName;
+		} else if ($tableName == self::GREYLIST) {
+			$table = $greyListTableName;
+		}
+		$sql = "SELECT * FROM " . $table;
+		$result = $conn->query($sql);
+		while($row = $result->fetch_assoc()) {
+			$idp = array();
+			$idp['timestamp'] = $row['date'];
+			$idp['entityid'] = $row['entityId'];
+			$idp['reason'] = $row['reason'];
+			array_push($listOfIdPs, $idp);
+		}
+		$conn->close();
+		return $listOfIdPs;
+	}
+
+	/**
+	 * Function returns array of all entityId in whitelist/greylist
+	 * @param string $tableName 'whitelist' or 'greylist'
+	 * @return array of entityIds
+	 */
+	public static function getAllEntityIds($tableName) {
+		$databaseConnector = new DatabaseConnector();
+		$conn = $databaseConnector->getConnection();
+		$whiteListTableName = $databaseConnector->getWhiteListTableName();
+		$greyListTableName = $databaseConnector->getGreyListTableName();
+		$table = null;
+		$listOfIdPs = array();
+		assert($conn != NULL);
+
+		if ($tableName == self::WHITELIST) {
+			$table = $whiteListTableName;
+		} else if ($tableName == self::GREYLIST) {
+			$table = $greyListTableName;
+		}
+		$sql = "SELECT * FROM " . $table;
+		$result = $conn->query($sql);
+		while($row = $result->fetch_assoc()) {
+			array_push($listOfIdPs, $row["entityId"]);
+		}
+		$conn->close();
+		return $listOfIdPs;
+	}
+
+
+	/**
+	 * Function inserts the line into table with $tableName
+	 * @param string $tableName  'whitelist' or 'greylist'
+	 * @param string $entityId
+	 * @param string $reason
+	 */
+	public static function insertTolist($tableName, $entityId, $reason) {
+		$databaseConnector = new DatabaseConnector();
+		$conn = $databaseConnector->getConnection();
+		$whiteListTableName = $databaseConnector->getWhiteListTableName();
+		$greyListTableName = $databaseConnector->getGreyListTableName();
+		$table = null;
+		assert($conn != NULL);
+
+		if ($tableName == self::WHITELIST) {
+			$table = $whiteListTableName;
+		} else if ($tableName == self::GREYLIST) {
+			$table = $greyListTableName;
+		}
+		$sql = "INSERT INTO " . $table . "(entityId, reason ) VALUES ('".$entityId."','" . $reason . "');";
```

Comment:
Use prepared statements instead, do not construct SQL query directly from the strings, it can be misused for SQL injection.